sources/core/Stride.Core/Threading/ThreadPool.cs

```diff
@@ -189,5 +256,36 @@ public void Dispose()
                 
             }
         }
+
+        unsafe struct Work
+        {
+            public object Data;
+            public delegate*<object, void> WorkHandler;
+        }
+
+        private interface ISemaphore : IDisposable
+        {
+            public void Release( int Count );
+            public void Wait( int timeout = - 1 );
+        }
+
+        private sealed class DotnetLifoSemaphore : ISemaphore
+        {
+            private readonly IDisposable semaphore;
+            private readonly Func<int, bool, bool> wait;
+            private readonly Action<int> release;
+
+            public DotnetLifoSemaphore(int spinCount)
+            {
+                Type lifoType = Type.GetType("System.Threading.LowLevelLifoSemaphore");
```

Comment:
Would be good to add a comment here that we use reflection to access an internal type